doc/source/whatsnew/v1.4.3.rst

```diff
@@ -24,6 +24,7 @@ Fixed regressions
 - Fixed regression in :func:`read_csv` with ``index_col=False`` identifying first row as index names when ``header=None`` (:issue:`46955`)
 - Fixed regression in :meth:`.DataFrameGroupBy.agg` when used with list-likes or dict-likes and ``axis=1`` that would give incorrect results; now raises ``NotImplementedError`` (:issue:`46995`)
 - Fixed regression in :meth:`DataFrame.resample` and :meth:`DataFrame.rolling` when used with list-likes or dict-likes and ``axis=1`` that would raise an unintuitive error message; now raises ``NotImplementedError`` (:issue:`46904`)
+- Fixed regression in :func:`assert_index_equal` when ``check_order=False`` and :class:`Index` has extension dtype (:issue:`47207`)
```

Comment:
I think I may have been wrong to add the regression label to #47207

Not strictly a regression since 1.3.5 did not have EA backed Indexes and the code sample created an object dtype Index. maybe this should be in the bugfix section. wdyt?

Reply:
You are correct wrt ea back indexes, but you can also trigger this with
```
idx1 = pd.Index([1, 3], dtype="object")
idx2 = pd.Index([3, 1], dtype="object")
tm.assert_index_equal(idx1, idx2, check_order=False)
```

I'll add a test and will adjust the release note

Reply:
Added both